cognite/client/data_classes/simulators/simulators.py

```diff
@@ -0,0 +1,517 @@
+        self.step_fields = step_fields
+        self.unit_quantities = unit_quantities
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            external_id=resource["externalId"],
+            name=resource["name"],
+            file_extension_types=resource["fileExtensionTypes"],
+            model_types=SimulatorModelType._load_list(resource["modelTypes"], cognite_client)
+            if "modelTypes" in resource
+            else None,
+            step_fields=SimulatorStep._load_list(resource["stepFields"], cognite_client)
+            if "stepFields" in resource
+            else None,
+            unit_quantities=SimulatorQuantity._load_list(resource["unitQuantities"], cognite_client)
+            if "unitQuantities" in resource
+            else None,
+        )
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        if isinstance(self.model_types, SimulatorModelType):
+            output["modelTypes" if camel_case else "model_types"] = self.model_types.dump(camel_case=camel_case)
+        if isinstance(self.step_fields, SimulatorStep):
+            output["stepFields" if camel_case else "step_fields"] = self.step_fields.dump(camel_case=camel_case)
+        if isinstance(self.unit_quantities, SimulatorQuantity):
+            output["unitQuantities" if camel_case else "unit_quantities"] = self.unit_quantities.dump(
+                camel_case=camel_case
+            )
+
+        return output
+
+
+class SimulatorWrite(SimulatorCore):
+    def __init__(
+        self,
+        external_id: str,
+        name: str,
+        file_extension_types: str | SequenceNotStr[str],
+        model_types: SimulatorModelType | Sequence[SimulatorModelType] | None = None,
+        step_fields: SimulatorStep | Sequence[SimulatorStep] | None = None,
+        unit_quantities: SimulatorQuantity | Sequence[SimulatorQuantity] | None = None,
+    ) -> None:
+        super().__init__(
+            external_id=external_id,
+            name=name,
+            file_extension_types=file_extension_types,
+            model_types=model_types,
+            step_fields=step_fields,
+            unit_quantities=unit_quantities,
+        )
+
+    def as_write(self) -> SimulatorWrite:
+        """Returns a writeable version of this resource"""
+        return self
+
+
+class Simulator(SimulatorCore):
+    def __init__(
+        self,
+        external_id: str,
+        name: str,
+        file_extension_types: str | SequenceNotStr[str],
+        created_time: int | None = None,
+        last_updated_time: int | None = None,
+        id: int | None = None,
+        model_types: SimulatorModelType | Sequence[SimulatorModelType] | None = None,
+        step_fields: SimulatorStep | Sequence[SimulatorStep] | None = None,
+        unit_quantities: SimulatorQuantity | Sequence[SimulatorQuantity] | None = None,
```

Comment:
Fields like id which are required in the API response, should not be optional here.